src/linked_list.py

```diff
@@ -0,0 +1,95 @@
+"""Implements a linked list class."""
+
+
+class LinkedList(object):
+    """A singly-linked list."""
+
+    def __init__(self, data=None):
+        """Create an instance of type LinkedList. Allow data to be passed in."""
+        self.head = None
+        self.tail = None
+        if data is not None:
+            try:
+                for item in data:
+                    if item is data[0]:
+                        self.head = Node(item, next=None)
+                        self.tail = self.head
+                    else:
+                        self.head = Node(item, self.head)
+            except TypeError:
+                node = Node(data, next=None)
+                self.head = node
+                self.tail = self.head
+
+
+    def push(self, val):
+        self.head = Node(val, self.head)
+
+
+    def pop(self):
+        if not self.head:
+            return None
+        val = self.head.data
+        if self.head.next == None:
+            self.tail = None
+        self.head = self.head.next
+        return val
+
+    def size(self):
+        count = 0
+        if self.head:
+            count = 1
+            cur = self.head
+            while cur.next:
+                count +=1
+                cur = cur.next
+            return count
+        return 0
```

Comment:
Why calculate the size on the fly everytime this method is called instead of just keeping track of the size on every push and pop?